oracle.go

```diff
@@ -280,8 +272,79 @@ func (o *Oracle) RotateRootCredentials(ctx context.Context, statements []string)
 	return o.RawConfig, nil
 }
 
+func (o *Oracle) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	rotateStatements := statements.Rotation
+	if len(rotateStatements) == 0 {
+		rotateStatements = []string{defaultRotateCredsSql}
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	variables := map[string]string{
+		"username": username,
+		"password": password,
+	}
+
+	queries := splitQueries(rotateStatements)
+	if len(queries) == 0 { // Extra check to protect against future changes
+		return "", "", errors.New("no rotation queries found")
+	}
+
+	// Lock the SQL connection
+	o.Lock()
+	defer o.Unlock()
+
+	db, err := o.getConnection(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to get database connection: %w", err)
+	}
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to create database transaction: %w", err)
+	}
+	defer tx.Rollback()
```

Comment:
What does this do in the success case (i.e. post-commit)? If the driver is turning it into a no-op, it would be worth a comment explaining that.

Reply:
Or moving it into the `if` might be clearer.

Reply:
Technically it errors: `sql: transaction has already been committed or rolled back` but this is effectively a no-op. I'll put in a comment explaining it.